### ISSUE TITLE
Validate support for istio 1.8

### ISSUE THREAD

**Issue by joekelley** [solo-io/gloo-mesh#1090]
This issue will track the status of Istio 1.8 support in Gloo Mesh.

- [x] [Istio sidecar crash](https://github.com/istio/istio/issues/28620) is still affecting our CI
- [x] ingress discovery should be updated to be compatible with Istio's recommended east-west gateway #1094

**Comment by harveyxia** [comment]
Unfortunately it looks like we're [still encountering the Envoy crash](https://github.com/solo-io/gloo-mesh/runs/1432226526#step:10:3201) in the 1.8 e2e tests. This is also confirmed by the [1.8 release notes](https://istio.io/latest/news/releases/1.8.x/announcing-1.8/change-notes/):

> Race condition with Envoy aggregate cluster when creating an EnvoyFilter and ServiceEntry for the same service. Istio-injected pods are unable to start up due to istio-proxy crashing with a segfault. See [Issue #28620](https://github.com/istio/istio/issues/28620) for more information.

**Comment by joekelley** [comment]
We need to update mesh discovery such that [default gateway label](https://github.com/solo-io/gloo-mesh/blob/d182c3d28e34b26a24a0c5dea5cb1bad5613f20d/pkg/common/defaults/istio.go#L6) and [default gateway port name](https://github.com/solo-io/gloo-mesh/blob/5d3ea22d594b3fbeb83ceb930f0c4459fedeadc6/pkg/mesh-discovery/translation/mesh/detector/istio/istio_mesh_detector.go#L38) are configurable by the user. This is because the Istio [multi-primary install docs](https://istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/#install-the-east-west-gateway-in-cluster1) guide the user to create an `istio-eastwestgateway` (vs. `istio-ingressgateway`) with an `mtls` port (vs. `tls`). This way east-west traffic flows through a different gateway than ingress traffic. example east-west gateway:

```yaml
apiVersion: v1
kind: Service
metadata:
  annotations: ...
  creationTimestamp: "2020-11-23T20:38:52Z"
  labels:
    app: istio-eastwestgateway
    install.operator.istio.io/owning-resource: eastwest
    install.operator.istio.io/owning-resource-namespace: istio-system
    istio: eastwestgateway
    istio.io/rev: default
    operator.istio.io/component: IngressGateways
    operator.istio.io/managed: Reconcile
    operator.istio.io/version: 1.8.0
    release: istio
    topology.istio.io/network: network1
  name: istio-eastwestgateway
  namespace: istio-system
  resourceVersion: "797"
  selfLink: /api/v1/namespaces/istio-system/services/istio-eastwestgateway
  uid: e97c74e0-6273-4686-acd7-de0ae62791ba
spec:
  clusterIP: 10.96.53.130
  externalTrafficPolicy: Cluster
  ports:
  - name: status-port
    nodePort: 31655
    port: 15021
    protocol: TCP
    targetPort: 15021
  - name: mtls
    nodePort: 32766
    port: 15443
    protocol: TCP
    targetPort: 15443
  - name: tcp-istiod
    nodePort: 32275
    port: 15012
    protocol: TCP
    targetPort: 15012
  - name: tcp-webhook
    nodePort: 30074
    port: 15017
    protocol: TCP
    targetPort: 15017
  selector:
    app: istio-eastwestgateway
    istio: eastwestgateway
    topology.istio.io/network: network1
  sessionAffinity: None
  type: NodePort
status:
  loadBalancer: {}
```

Proposal: we could add these values to the setting CRDs, as global defaults and/or per-cluster overrides with some fallback.

**Comment by joekelley** [comment]
Resolved with the release of istio 1.8.3 https://istio.io/latest/news/releases/1.8.x/announcing-1.8.3/#changes
